api/src/test/java/com/epam/pipeline/acl/pipeline/PipelineApiServiceTest.java

```diff
@@ -0,0 +1,831 @@
+
+        assertThrows(AccessDeniedException.class, () -> pipelineApiService.delete(ID, true));
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldDenyDeletePipelineForUserRole() {
+        initAclEntity(pipeline, AclPermission.WRITE);
+        doReturn(pipeline).when(mockPipelineManager).delete(ID, true);
+
+        assertThrows(AccessDeniedException.class, () -> pipelineApiService.delete(ID, true));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadAllRunsByPipelineForAdmin() {
+        doReturn(pipelineRunList).when(mockPipelineRunManager).loadAllRunsByPipeline(ID);
+
+        assertThat(pipelineApiService.loadAllRunsByPipeline(ID)).isEqualTo(pipelineRunList);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldLoadAllRunsByPipelineWhenPermissionIsGranted() {
+        initAclEntity(pipeline, AclPermission.READ);
+        initAclEntity(pipelineRun, AclPermission.READ);
+        doReturn(pipelineRunList).when(mockPipelineRunManager).loadAllRunsByPipeline(ID);
+
+        final List<PipelineRun> returnedPipelineRunList = pipelineApiService.loadAllRunsByPipeline(ID);
+
+        assertThat(returnedPipelineRunList).hasSize(1).contains(pipelineRun);
+        assertThat(returnedPipelineRunList.get(0).getMask()).isEqualTo(READ_PERMISSION);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldDenyLoadAllRunsByPipelineWhenPermissionIsNotGranted() {
+        initAclEntity(pipeline);
+        doReturn(pipelineRunList).when(mockPipelineRunManager).loadAllRunsByPipeline(ID);
+
+        assertThrows(AccessDeniedException.class, () -> pipelineApiService.loadAllRunsByPipeline(ID));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadVersionFromGitForAdmin() throws GitClientException {
+        doReturn(revisionList).when(mockVersionManager).loadAllVersionFromGit(ID);
+
+        assertThat(pipelineApiService.loadAllVersionFromGit(ID)).isEqualTo(revisionList);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldLoadAllVersionFromGitWhenPermissionIsGranted() throws GitClientException {
+        initAclEntity(pipeline, AclPermission.READ);
+        doReturn(revisionList).when(mockVersionManager).loadAllVersionFromGit(ID);
+
+        assertThat(pipelineApiService.loadAllVersionFromGit(ID)).isEqualTo(revisionList);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldDenyLoadAllVersionFromGitWhenPermissionIsNotGranted() throws GitClientException {
+        initAclEntity(pipeline);
+        doReturn(revisionList).when(mockVersionManager).loadAllVersionFromGit(ID);
+
+        assertThrows(AccessDeniedException.class, () -> {
+            try {
+                pipelineApiService.loadAllVersionFromGit(ID);
+            } catch (GitClientException e) {
+                e.printStackTrace();
+            }
+        });
+    }
```

Comment:
Probably these tests as well as all others depending on `PipelineVersionManager` can be moved to `PipelineApiServiceGitTest`.